Automattic/wp-calypso
# Plugins: Improve performance of plugin selectors

#### Proposed Changes

This fixes some performance issues with the plugins selectors by reducing runtime complexity, indexing, and memoizing where possible.

#### Testing Instructions



1. Visit `/plugins/manage` in your dev environment and compare it to what you see in production. There should be no difference.
2. Visit /plugins/manage as an agency with many sites. See this Slack comment for an example: p1669769472607889-slack-C039K7CD119. Verify that the client is responsive after the initial network calls are completed. If the agency has more than 20 plugins then note that the icons and titles from the plugin repository will not show.
3. Do regression testing on the other plugins pages: `/plugins/manage/:site` and `/plugins/:plugin_slug`

#### Pre-merge Checklist





Related to #
